include/deal.II/base/symmetric_tensor.h

```diff
@@ -580,12 +580,20 @@ class SymmetricTensor
   SymmetricTensor (const SymmetricTensor<rank,dim,OtherNumber> &initializer);
 
   /**
+   * Assignment operator from symmetric tensors with different underlying scalar type.
+   * This obviously requires that the @p OtherNumber type is convertible to @p
+   * Number.
+   */
+  template <typename OtherNumber>
+  SymmetricTensor &operator = (const SymmetricTensor<rank,dim,OtherNumber> &rhs);
+
+  /**
    * This operator assigns a scalar to a tensor. To avoid confusion with what
    * exactly it means to assign a scalar value to a tensor, zero is the only
    * value allowed for <tt>d</tt>, allowing the intuitive notation
    * <tt>t=0</tt> to reset all elements of the tensor to zero.
    */
-  SymmetricTensor &operator = (const Number d);
```

Comment:
This is `SymmetricTensor`, so no ;-) But I don't think that changing this to a pass by reference is would stop this working.